protocol/messenger_communities.go

```diff
@@ -1921,3 +1914,389 @@ func (m *Messenger) RequestExtractDiscordChannelsAndCategories(filesToImport []s
+						continue
+					}
+					authorAvatarUrls[discordMessage.Author.Id] = discordMessage.Author.AvatarUrl
+				}
+
+				payload, err := proto.Marshal(discordMessage)
+				if err != nil {
+					m.logger.Error("failed to marshal discord message", zap.Error(err))
+					importProgress.AddTaskError(discord.ImportMessagesTask, discord.Warning(discord.ErrMarshalMessage.Error()))
+					importProgress.UpdateTaskProgress(discord.ImportMessagesTask, progressValue)
+					continue
+				}
+
+				// Creating a `common.RawMessage` to get a `messageID`
+				rawMessage := &common.RawMessage{
+					Payload:        payload,
+					MessageType:    protobuf.ApplicationMetadataMessage_CHAT_MESSAGE,
+					Sender:         discordCommunity.PrivateKey(),
+					SkipEncryption: true,
+				}
+				messageID, err := m.sender.GetMessageID(rawMessage)
+				if err != nil {
+					m.logger.Error("failed to get message ID", zap.Error(err))
+					importProgress.AddTaskError(discord.ImportMessagesTask, discord.Warning(err.Error()))
+					importProgress.UpdateTaskProgress(discord.ImportMessagesTask, progressValue)
+					continue
+				}
+
+				// Convert timestamp to unix timestamp
+				discordMessage.Timestamp = fmt.Sprintf("%d", timestamp.Unix())
+
+				if discordMessage.TimestampEdited != "" {
+					timestampEdited, err := time.Parse(layout, discordMessage.TimestampEdited)
+					if err != nil {
+						m.logger.Error("failed to parse discord message timestamp", zap.Error(err))
+						importProgress.AddTaskError(discord.ImportMessagesTask, discord.Warning(err.Error()))
+						importProgress.UpdateTaskProgress(discord.ImportMessagesTask, progressValue)
+						continue
+					}
+					// Convert timestamp to unix timestamp
+					discordMessage.TimestampEdited = fmt.Sprintf("%d", timestampEdited.Unix())
+				}
+
+				communityPubKey := discordCommunity.PrivateKey().PublicKey
+				messageToSave := &common.Message{
+					ID:               types.EncodeHex(messageID),
+					WhisperTimestamp: uint64(timestamp.Unix()),
+					From:             types.EncodeHex(crypto.FromECDSAPub(&communityPubKey)),
+					Seen:             true,
+					LocalChatID:      processedChannelIds[channel.Channel.ID],
+					SigPubKey:        &communityPubKey,
+					CommunityID:      discordCommunity.IDString(),
+					ChatMessage: protobuf.ChatMessage{
+						MessageType: protobuf.MessageType_COMMUNITY_CHAT,
+						ContentType: protobuf.ChatMessage_DISCORD_MESSAGE,
+						Clock:       m.getTimesource().GetCurrentTime(),
+						ChatId:      processedChannelIds[channel.Channel.ID],
+						Payload: &protobuf.ChatMessage_DiscordMessage{
+							DiscordMessage: discordMessage,
+						},
+					},
+				}
+				discordMessagesToSave = append(discordMessagesToSave, discordMessage)
+				processedMessages = append(processedMessages, messageToSave)
+			}
+			importProgress.UpdateTaskProgress(discord.ImportMessagesTask, float32(len(processedMessages))/float32(totalMessageCount)*0.90)
```

Comment:
Oh here, I'm multiplying by `0.9` so the progress bar goes to 90%. Reason being, we still need to save the processed messages which will take time. So we don't want it to be 100% yet.